notes/03-data-processing/03b-quiz-recursions.MD

```diff
@@ -0,0 +1,74 @@
+## Quiz recursions
+
+### Which of these datastructures is recursive?
+
+```scala
+case class Person(name: String, age: Int, email: Option[String])
+
+case class Order(quantity: Int, price: Double, previousOrder: Option[Order])
+
+sealed trait MovingStorage
+case class Box(width: Int, height: Int) extends MovingStorage
+case class Truck(foot: Int) extends MovingStorage
+```
+
+1. Person
+1. (c) Order
+1. MovingStorage
+1. Option
+1. (c) java.io.File
```

Comment:
This one is confusing. Despite the fact that you could work with `java.io.File` in a recursive manner, it doesn't look like a data structure at all. You wouldn't get child elements from it without performing some IO.

Reply:
That's true, a `java.io.File` is not a recursive ADT, but in the video about recursions, I spend quite a lot of time talking about the recursive nature of files. The other example I used in the lecture is JSON. Maybe less confusing.

Reply:
Yeah, JSON would be much less confusing.